### PR TITLE
Parse generated YAML file

### PR DESCRIPTION
**Pull request by Jacob-Allebach**
# Parse generated YAML file

## Description

This pull request adds a function that uses PyYaml to parse the information from a given YAML file and returns the stored data so that it can be used later.

## Type of Change

- [x] Feature
- [ ] Bug fix
- [ ] Documentation

## Contributors

- @Jacob-Allebach 

## Reminder

 - All GitHub Actions should be in a passing state before any pull request is merged.
 - All PRs must be reviewed by at least one team member and one member of the Integration team!
 - Any issues this PR closes are tagged in the description!

### CODE REVIEW

**Review by PaigeCD — CHANGES_REQUESTED** [review]
Hi, Jacob. I have some quick changes to request before getting into the nitty-gritty of the code:
- Be sure to resolve the file's conflicts.
- Please merge main into your branch to get the poetry file and workflows!

File: gatorgrade/input/in_file_path.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import yaml
+
+#Parses a YAML file and returns its contents as a list of dictionaries
+def parse_yaml_file(file_path):
+    with open(file_path) as f:
+        data = yaml.load_all(f, Loader=yaml.FullLoader)
+        data_holder = []
+        for item in data: #Transfers the data from the file's generator to a list for later access
+            data_holder.append(item)
+        return data_holder
```
Comment (noorbuchi) [inline gatorgrade/input/in_file_path.py:10]: I think this code can be simplified by doing this:
```suggestion
        return list(data)
```

**Review by noorbuchi — CHANGES_REQUESTED** [review]
Hello! Thanks for your work in  setting this up. Here are few suggestions.

**Comment by noorbuchi** [comment]
Note that the build is failing, can you please investigate this and try to fix it before merging. Thanks!

**Comment by mariakimheinert** [comment]
Hello @Jacob-Allebach, just want to check in with you about this PR since it has now been open for 8 days. It appears that the code you have included in this PR does not process the data that is read from the `gatorgrade.yml` file into the "list of tuples" format that your team discussed. It is my understanding that @tuduun is expecting this formatted data for the function he is currently writing. Do you plan to add additional changes to this PR or should it be reviewed?

**Comment by Jacob-Allebach** [comment]
> Hello @Jacob-Allebach, I just want to check in with you about this PR since it has now been open for 8 days. It appears that the code you have included in this PR does not process the data that is read from the `gatorgrade.yml` file into the "list of tuples" format that your team discussed. It is my understanding that @tuduun is expecting this formatted data for the function he is currently writing. Do you plan to add additional changes to this PR or should it be reviewed?

Yes, this code just parses the YAML file and gets the raw data. I'm planning on making a separate pull request for the other function I'm writing that will format that data into the list of tuples, but it's not completed quite yet. If you think that should also be included in this pull request, I would be fine with adding it to this one once that is completed.

**Review by noorbuchi — APPROVED** [review]
As long as the team is on the same page regarding this PR and has a plan to use the function you've included here, then I think it's good to merge this. However, please make sure to address all questions brought up by @mariakimheinert before merging. Thanks for all your work!

**Comment by mariakimheinert** [comment]
@Jacob-Allebach, I agree with @noorbuchi. It's fine to merge this function in first and then open another PR for the other function you plan to write. @laurennevill, @Michionlion, @PaigeCD, @LiamBlack3, would you please review this PR at your earliest convenience?

**Review by PaigeCD — APPROVED** [review]

**Review by Michionlion — APPROVED** [review]
This looks good, great work!

**Pull request merged**